torch/onnx/utils.py

```diff
@@ -331,41 +331,34 @@ def _trace_and_get_graph_from_model(model, args):
     return trace_graph, torch_out
 
 
-def _model_to_graph(model, args, verbose=False,
-                    input_names=None, output_names=None,
-                    operator_export_type=OperatorExportTypes.ONNX,
-                    example_outputs=None, propagate=False,
-                    _retain_param_name=False, do_constant_folding=True,
-                    _disable_torch_constant_prop=False, fixed_batch_size=False, training=None):
-    from torch.onnx.symbolic_helper import _export_onnx_opset_version
-    # Special case for common case of passing a single Tensor
-    if isinstance(args, torch.Tensor):
-        args = (args, )
-
-    if isinstance(example_outputs, torch.Tensor):
-        example_outputs = [example_outputs]
-
+def _model_to_jit_graph(model, args, _retain_param_name, update_jit_scripting_passes):
     torch_out = None
     if isinstance(model, torch.jit.ScriptModule):
-        assert example_outputs is not None, "example_outputs must be provided when exporting a ScriptModule"
         try:
             graph = model.forward.graph
             torch._C._jit_pass_onnx_function_substitution(graph)
-            method_graph, params = torch._C._jit_pass_lower_graph(graph, model._c)
+            if not update_jit_scripting_passes:
+                method_graph, params = torch._C._jit_pass_lower_graph(graph, model._c)
+            else:
+                freezed_m = torch._C._freeze_module(model._c)
+                method_graph = freezed_m._get_method('forward').graph
+                method_graph.eraseInput(0)  # Remove 'self' from model inputs
+                params = []
+
+                torch._C._jit_pass_onnx_preprocess(method_graph)
```

Comment:
The preprocess pass was designed to be applied on both tracing and scripting, and at later stage of the conversion (just before converting all nodes to onnx namespace). On the other hand, I see the necessity of putting this pass before remove_mutation. Maybe it should be put into a separate pass, one that focus on remove mutation for onnx. What do you think?

Reply:
So my idea is to eventually move these two passes to the main code-path and replace remove_inplace_ops pass.
I was thinking we can run the preprocess pass twice for now.

Reply:
So the an exception msg with string.format appears in the graph as RaiseException node with a tuple which contains aten::format. So I'm adding this op type to the allowed list to enable lower_tuple pass.
This unblocks scripting a number of tests on pytorch_onnx_onnxruntime.